src/cmd/troubleshoot/dynakube.go

```diff
@@ -59,31 +96,37 @@ func checkDynakubeCrdExists(troubleshootCtx *troubleshootContext) error {
 	dynakubeList := &dynatracev1beta1.DynaKubeList{}
 	err := troubleshootCtx.apiReader.List(troubleshootCtx.context, dynakubeList, &client.ListOptions{Namespace: troubleshootCtx.namespaceName})
 
-	if runtime.IsNotRegisteredError(err) {
-		return errorWithMessagef(err, "CRD for Dynakube missing")
-	} else if err != nil {
-		return errorWithMessagef(err, "could not list Dynakube")
+	if err != nil {
+		if runtime.IsNotRegisteredError(err) {
+			err = errors.Wrap(err, "CRD for Dynakube missing")
+		} else {
+			err = errors.Wrap(err, "could not list Dynakube")
+		}
+		return err
 	}
 
 	logInfof("CRD for Dynakube exists")
 	return nil
```

Comment:
unwrapping `if`s for better readbility
```suggestion
    if err == nil {    
        logInfof("CRD for Dynakube exists")
    	return nil
    }
	
	if runtime.IsNotRegisteredError(err) {
		err = errors.Wrap(err, "CRD for Dynakube missing")
	} else {
		err = errors.Wrap(err, "could not list Dynakube")
	}
	
	return err
```

Reply:
Typically I'd agree but reverting idiomatic `if err != nil` to `if err == nil` may lead to problems because it's easy to overlook inverted logic.

Reply:
then I'd suggest refactoring the inner code so its not that deep, e.g.:
```suggestion
	if err != nil {
		return determineDynakubeError(err)
	}

	logInfof("CRD for Dynakube exists")
	return nil
	...
}

func determineDynakubeError(err error) error {
       if runtime.IsNotRegisteredError(err) {
	    return errors.Wrap(err, "CRD for Dynakube missing")
        } else {
	    return errors.Wrap(err, "could not list Dynakube")
	}
}	
```